### ISSUE TITLE
Incompatible type on Map field initializer

### ISSUE THREAD

**Issue by Cor-cor** [HaxeFoundation/intellij-haxe#586]
Probably this issue is somewhat similar to #388.

![idea_2017-03-28_10-58-54](https://cloud.githubusercontent.com/assets/9751023/24394681/991adf70-13a5-11e7-864c-ca18d755d5e6.png)

Map initializer for class field is parsed as `Array<Dynamic>`, which is wrong. But parsing works fine if initialization code is moved to method body.

Also note how at first `map1` is of type `Array<Dynamic>`:
![idea_2017-03-28_11-08-54](https://cloud.githubusercontent.com/assets/9751023/24395772/71c0eee8-13a9-11e7-9c62-dc5aa027c00b.png)
and then the correct type inference kicks in:
![idea_2017-03-28_11-09-25](https://cloud.githubusercontent.com/assets/9751023/24395810/92838320-13a9-11e7-9287-057e398dcd6b.png)

Plugin version 0.10.1.1 on IDEA 2016.3.5.

Repro:
```
package ;

class Test {

    var map1 = [0 => "zero", 1 => "one"];
    var map2:Map<Int, String> = [0 => "zero", 1 => "one"];

    public function new() {}
}
```

**Comment by EricBishton** [comment]
This is a semantic annotation error.  It parses cleanly on release 0.11.2.

**Comment by Netforza** [comment]
The problem persists on plugin version 0.11.2 on IDEA 2017.3.4
![haxeplugin_error](https://user-images.githubusercontent.com/12975310/35851161-cabd4dc4-0b2f-11e8-85ec-7ac941bbdecc.png)

**Comment by EricBishton** [comment]
If you'd like to take a crack at it, the code is right here: https://github.com/HaxeFoundation/intellij-haxe/blob/abf257594f7d8c00fff08b0849d97bfb8aae159c/src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java#L87

**Comment by EricBishton** [comment]
Fixed with PR #753.  Waiting for code review.

**Comment by EricBishton** [comment]
Submitted with https://github.com/HaxeFoundation/intellij-haxe/commit/258bfcf5b9d9158d5b1af9dfbf27754978f07953
